core/src/test/scala/org/apache/spark/rpc/akka/AkkaRpcEnvSuite.scala

```diff
@@ -47,4 +56,60 @@ class AkkaRpcEnvSuite extends RpcEnvSuite {
     }
   }
 
+  test("timeout on ask Future with RpcTimeout") {
+
+    class EchoActor(sleepDuration: Long) extends Actor {
+      def receive: Receive = {
+        case msg =>
+          Thread.sleep(sleepDuration)
+          sender() ! msg
+      }
+    }
+
+    val system = ActorSystem("EchoSystem")
+    val echoActor = system.actorOf(Props(new EchoActor(0)), name = "echo")
+    val sleepyActor = system.actorOf(Props(new EchoActor(50)), name = "sleepy")
+
+    val shortProp = "spark.rpc.short.timeout"
+    val timeout = new RpcTimeout(10 millis, shortProp)
+
+    try {
+
+      // Ask with immediate response
+      var fut = echoActor.ask("hello")(timeout.duration).mapTo[String].
+        recover(timeout.addMessageIfTimeout)
+
+      // This should complete successfully
+      val result = timeout.awaitResult(fut)
+
+      assert(result.nonEmpty)
+
+      // Ask with delayed response
+      fut = sleepyActor.ask("goodbye")(timeout.duration).mapTo[String].
+        recover(timeout.addMessageIfTimeout)
+
+      // Allow future to complete with failure using plain Await.result, this will return
+      // once the future is complete
+      val msg1 =
+        intercept[RpcTimeoutException] {
+          Await.result(fut, 200 millis)
```

Comment:
I noticed this test sometimes outputs a message that there is a dead letter from the ask message that times out.  Is that something we need to fix?  I looked but couldn't find a simple way to stop the dead letter logging.